### DISCUSSION TITLE
Watchlist: Interesting new Models / Papers / Tech

### DISCUSSION THREAD

**Discussion by Acly** [Acly/krita-ai-diffusion#779] [Ideas]
This is a list of AI models, papers and ComfyUI nodes which might be candidates for integration into the plugin.

### Guide to posting feature suggestions and requests:

* **You found out about a project or new model? You want to post a link because you think it will make a good addition?**
  * You're welcome to post links here in this discussion.
* **You want to share an evaluation or image comparisons?**
  * Feel free to open a separate discussion topic about your results .
* **You want to contribute by implementing a new feature (in code)?**
  * It's a good idea to create an "enhancement" issue to let everyone know.

> [!IMPORTANT]
>
> **Please do not open separate issues / discussions just to post a link!**
>
> This helps avoid clutter. Keep in mind there are many shiny new things coming out all the time, and realistically only a fraction of them will be integrated.

### How to make it happen?

In order of most likely success:
1. Implement it! (If you're unsure, discuss it first)
2. Evaluate and compare techniques (eg in ComfyUI), share images and workflows. It's a significant portion of the work, so it helps!
3. Comment/upvote in this discussion. _One comment per feature allows people to thumbs-up to show support._

### Base Models

| Project | Architecture | Params | License | Comment |
|---------|--------------|------------|------|------|
| [Kolors](https://huggingface.co/Kwai-Kolors/Kolors) | U-Net, GLM | 3.5B | Non-commercial | Inpaint, IP-Adapter |
| [PixArt-Σ](https://github.com/PixArt-alpha/PixArt-sigma) | DiT, T5 | 0.6B | Open RAIL++-M | |
| [Hunyuan-DiT](https://huggingface.co/Tencent-Hunyuan/HunyuanDiT) | DiT, T5 | 1.5B | Tencent-Hunyuan (limited) |
| [AuraFlow](https://huggingface.co/fal/AuraFlow) | DiT/Flow, T5 | 6.8B | Apache 2.0 | Beta 0.1 version |
| [UltraPixel](https://jingjingrenabc.github.io/ultrapixel/) | Cascade, CLIP | 1B/3.5B + 0.7B/1.5B | Apache 2.0 | |
| [Flux [dev]](https://huggingface.co/black-forest-labs/FLUX.1-dev) | DiT/Flow, T5 | 12B |  Non-commercial | ✅ [v1.22.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.22.0) (experimental) |
| [Flux [schnell]](https://huggingface.co/black-forest-labs/FLUX.1-dev) | DiT/Flow, T5 | 12B |  Apache 2.0 | ✅ [v1.22.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.22.0) (experimental) |

### Control Layer

| Project | Description |  Comment |
|---------|--------------|------------|
| [MistoLine / AnyLine](https://github.com/TheMistoAI/MistoLine) | Line art detector and ControlNet | ✅ [v1.19.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.19.0) | 
| [Xinsir CN models](https://huggingface.co/xinsir/controlnet-scribble-sdxl-1.0) | Scribble, OpenPose, Canny for SDXL | ✅  [v1.19.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.19.0) |
| [IC-Light](https://github.com/lllyasviel/IC-Light) | Lighting/shading control | requires Layer-Diffusion | 
| [PuLID](https://github.com/ToTheBeginning/PuLID) | Face adapter | | 
| [Recolor](https://huggingface.co/stabilityai/control-lora) | SDXL Control Lora | | 
| [PhotoMaker](https://github.com/TencentARC/PhotoMaker) | Face adapter (SDXL) | #388 No insightface |

### Performance

| Project | Description |  Comment |
|---------|--------------|------------|
| [Trajectory Consistency Distillation (TCD)](https://github.com/jabir-zheng/TCD) | Disstilled LoRA models + sampler for low-step (SD1.5, SDXL) | ❌ LoRA, ❓ Sampler  | 
| [Phased Consistency Model (PCM)](https://github.com/G-U-N/Phased-Consistency-Model/) | Distilled LoRA models (SD1.5, SDXL), allows CFG  | [Models](https://huggingface.co/Kijai/converted_pcm_loras_fp16/tree/main), already works |
| [Distribution Matching Distillation (DMD2)](https://github.com/tianweiy/DMD2) | Distilled 4-step LoRA models (SD1.5, SDXL) | ❌ Prefer Hyper-SD |
| [Automatic CFG](https://github.com/Extraltodeus/ComfyUI-AutomaticCFG) | Rescaled CFG, can disable CFG to boost performance | | 
| [Hyper-SD](https://hyper-sd.github.io/) | Distilled 1/4/8-step LoRA models, allows CFG | ✅ [v1.19.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.19.0) | 
| [HiDiffusion](https://hidiffusion.github.io/) | High resolutions w/o 2-pass at improved speed | #651 [ComfyUI Node](https://github.com/nullquant/ComfyUI-BrushNet) |
| [Align-Your-Steps (AYS)](https://research.nvidia.com/labs/toronto-ai/AlignYourSteps/) | Custom scheduler for low-step | ✅ [v1.20.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.20.0) |
| [T-GATE](https://github.com/HaozheLiu-ST/T-GATE) | Attention caching |  | 
| [Improved ToMe (ToDo)](https://github.com/ethansmith2000/ImprovedTokenMerge) | Token merging | | 

### Other

| Project | Description |  Comment |
|---------|--------------|-------------|
| [Layer-Diffusion](https://github.com/huchenlei/ComfyUI-layerdiffuse) | Alpha channel for image generation | #453 | 
| [BrushNet](https://github.com/TencentARC/BrushNet) | Inpaint extension model (SD1.5, SDXL)  | [ComfyUI node](https://github.com/nullquant/ComfyUI-BrushNet) |
| [ToonCrafter](https://doubiiu.github.io/projects/ToonCrafter/) | Animation keyframe interpolation | [ComfyUI node](https://github.com/kijai/ComfyUI-DynamiCrafterWrapper/)
| [SUPIR](https://github.com/Fanghua-Yu/SUPIR) | Upscale model, diffusion |  [ComfyUI node](https://github.com/kijai/ComfyUI-SUPIR) |
| [APISR](https://github.com/Kiteretsu77/APISR) | Upscale model, transformer  |  [ComfyUI node](https://github.com/ZHO-ZHO-ZHO/ComfyUI-APISR) |

**Comment by syl1130** [comment]
nice work,please

**Comment by moonwhaler** [comment]
Vote: Control Layer - Xinsir CN models

**Comment by moonwhaler** [comment]
Vote: Other - Layer-Diffusion

**Comment by Rogal80** [comment]
Vote: MistoLine

**Comment by Rogal80** [comment]
Vote: Improved ToMe

**Comment by Towering-Toska** [comment]
Vote: Layer-Diffusion. Layer-Diffusion sounds exactly like what I would now like the most after using Krita-AI for a night. I wasn't aware of this type of plugin, I wonder if we can already manually install and use that with the Krita plugin.

**Comment by DaVinchik** [comment]
We need supir!

**Comment by xingewh** [comment]
InstantID。InstantID seems to perform better than FaceID. 
(https://github.com/InstantID/InstantID)

**Reply by dkqjhoifh** [reply]
yes

**Comment by ETCore7** [comment]
Vote: Other - Tooncrafter. The video shows interesting examples of multiple frames generated between two key frames. Natural and consistent.

**Comment by Drakosha405** [comment]
what do you think about this. comparable to individual models or not, if the results are good then it can save a lot of space on the ssd.
https://huggingface.co/xinsir/controlnet-union-sdxl-1.0

**Reply by Acly** [reply]
Yes it sounds great. According to instruction you specify an ID depending on which mode you want, which isn't possible yet in ComfyUI as far as I know. It works without doing it, but maybe results can be better if you provide it.

**Comment by Corruptinator** [comment]
Not sure if this would be an interesting candidate to implement into the plugin but any chance the AuraFlow Ai could be implemented?
https://huggingface.co/fal/AuraFlow?ref=blog.fal.ai

**Comment by moonwhaler** [comment]
@Acly: This seems pretty interesting: background removal using InSPyReNet (https://github.com/plemeri/InSPyReNet).
There's also a Comfy Node for that: https://github.com/john-mnz/ComfyUI-Inspyrenet-Rembg?tab=readme-ov-file

I am already using the "transparent background" implementation (https://github.com/plemeri/transparent-background), but having this in Krita directly could be pretty interesting. 

Although, I admit this might be a bit different from the diffusion implementation you have done with Krita Diffusion.

**Comment by Drakosha405** [comment]
https://huggingface.co/blog/quanto-diffusers

good development and hopefully optimized versions of Aura Flow and ControlNet versions for it will appear soon. and then it is quite possible to consider adding it to the plugin

**Comment by waynemadsen** [comment]
https://huggingface.co/black-forest-labs/FLUX.1-dev

I'm sure flux support is on your radar and I look forward to building in krita with it!

**Comment by Acly** [comment]
Added a table for base models: Kolors, PixArt, Hunyuan-DiT, AuraFlow, Flux

They come up a lot, but none of them have any kind of ecosystem or commitment from open-source community so far. Kolors is the only one with inpaint and IP-Adapter support (both coming from the model authors directly).

**Comment by Marksusu** [comment]
https://huggingface.co/2vXpSwA7/iroiro-lora/blob/main/test_controlnet2/CN-posetest_v1.safetensors
https://x.com/nana_tsukisuwa/status/1832544135766769960
![GW5f81sbwAAJpwP](https://github.com/user-attachments/assets/882b392d-8f5c-4d58-b672-11fa278957b1)

这个可以用手绘简笔骨骼图生成人物动作

**Comment by ibow1983** [comment]
[Flux [dev]!! ControlNet for SDXL！！We need them!!!

**Reply by syl1130** [reply]
https://huggingface.co/Shakker-Labs/FLUX.1-dev-ControlNet-Union-Pro ,
https://huggingface.co/XLabs-AI/flux-controlnet-collections  
and https://huggingface.co/XLabs-AI/flux-ip-adapter

**Reply by waynemadsen** [reply]
I'd prefer to see a GGUF/unet implementation native in Krita before ControlNet on flux. But that's just my 8GB of VRAM crying in a corner...

**Comment by eglepere** [comment]
I'd really like to see SUPIR. The current upscale implementation feels lacking

**Comment by Leroy-X** [comment]
Vote: Base Models - Kolors

**Comment by waynemadsen** [comment]
flux controlnet upscaler: 
https://huggingface.co/jasperai/Flux.1-dev-Controlnet-Upscaler
same user has a good depth and normals map controlnet for flux as well.

**Comment by RanjithGadde447** [comment]
vote SUPIR UPSCALE

**Comment by waynemadsen** [comment]
FLUX ipadapter would be a great addition. It doesn't work as strongly as the style ipadapter for SDXL, but anything is better than the way FLUX resists 'style by' prompts.

https://huggingface.co/XLabs-AI/flux-ip-adapter

I suggest reading this paper: https://sandner.art/prompting-art-and-design-styles-in-flux-in-forge-and-comfyui/

**Comment by Andree11gor** [comment]
Vote: **IC-Light** - Lighting/shading control

**Comment by bingutech** [comment]
你好，这个插件能否基于Android的krita使用呢？或者通过重写的方式，期待你的回复，有机会希望我们能够合作，谢谢。bingutechnology@ gmail.com

**Comment by nesmonmehdydeal** [comment]
Can you add Z-IMAGE?

**Reply by Acly** [reply]
#2188

**Comment by fukc-gihtub** [comment]
Now this an upscaler we totally must have in krita-ai-diffusion:
https://www.reddit.com/r/StableDiffusion/comments/1pi2pxu/
I tried it, and it's _way_ better than the 4 upscalers (bad, bad, bad, and very bad) we currently have.
